web-ui/src/pages/FeedbackRequestPage.jsx

```diff
@@ -72,60 +76,98 @@ function getTemplates() {
+  const dueQuery = query.due?.toString();
+
+  const getStep = useCallback(() => {
+    if(!stepQuery || stepQuery < 1 || !/^\d+$/.test(stepQuery))
+      return 1;
+    else return parseInt(stepQuery);
+  },[stepQuery]);
 
-  let sendDate = query?.sendDate ? query.sendDate: todayDate.toString();
-  let dueDate = query?.dueDate ? query.dueDate: null
-  let activeStep = location?.search ? parseInt(stepQuery) : 1;
-  const numbersOnly = /^\d+$/.test(stepQuery);
+  const activeStep = getStep();
 
   const [preview, setPreview] = useState({open: false, selectedTemplate: null});
 
   const handlePreviewOpen = (event, selectedTemplate) => {
+    event.stopPropagation();
     setPreview({open: true, selectedTemplate: selectedTemplate});
-  }
+  };
 
   const handlePreviewClose = (selectedTemplate) => {
     setPreview({open: false, selectedTemplate: selectedTemplate});
-  }
+  };
 
-  const getFeedbackArgs = (step) => {
-    const nextQuery = {
-      ...query,
-      step: step
-    }
-
-    return `/feedback/request/?${queryString.stringify(nextQuery)}`;
+  const onCardClick = (template) => {
+    history.push(`/feedback/request/?template=${template.id}`);
   }
 
-  const handleQueryChange = (key, value) => {
-    let newQuery = {
-      ...query,
-      [key]: value
+  const hasTemplate = useCallback(() => {
+    return !!templateQuery;
+  }, [templateQuery])
+
+  const hasFrom = useCallback(() => {
+    return !!fromQuery;
+  }, [fromQuery])
+
+  const isValidDate = useCallback((dateString) => {
+    const timestamp = Date.parse(dateString);
+    return !isNaN(timestamp);
+  }, []);
+
+  const hasDue = useCallback(() => {
+    return (dueQuery && isValidDate(dueQuery))
+  }, [dueQuery, isValidDate]);
+
+  const canProceed = useCallback(() => {
+    switch(activeStep) {
+      case 1:
+        return hasTemplate();
+      case 2:
+        return hasTemplate() && hasFrom();
+      case 3:
+        return hasTemplate() && hasFrom() && hasDue();
+      default:
+        return false;
     }
-    history.push({...location, search: queryString.stringify(newQuery)});
-  }
-
-  const onCardClick = (template) => {
-    if (template.isAdHoc) {
-      setPreview({open: true, selectedTemplate: template});
-    } else {
-      console.log(`Selected ${template.title}`);
+  }, [activeStep, hasTemplate, hasFrom, hasDue]);
+
+  const handleSubmit = () => {};
+
+  const onNextClick = useCallback(() => {
+    if(!canProceed()) return;
+    if(activeStep === steps.length) handleSubmit();
+    query.step = activeStep+1;
+    history.push({...location, search: queryString.stringify(query)});
+  },[canProceed, activeStep, steps.length, query, location, history]);
+
+  const urlIsValid = useCallback(() => {
+    switch (activeStep) {
+      case 1:
+        return true;
+      case 2:
+        return hasTemplate();
+      case 3:
+        return hasTemplate() && hasFrom();
+      case 4:
+        return hasTemplate() && hasFrom() && hasDue();
```

Comment:
I don't think `case 4` will ever be reached, since `activeStep` only goes to 3 before we reroute the user to the confirmation page.

Reply:
This does bring up the issue that we have no validation for reaching the confirmation page at `/feedback/request/confirmation`. Maybe that should be a separate issue? I'm guessing that's why `handleSubmit` is left blank for now

Reply:
Agreed, case 4 can be removed. I also don't think I would try to address the submit in this issue. That should definitely be split.

Reply:
Ok, that validation will be addressed in issue #1077